src/RealTimeClient.js

```diff
@@ -180,15 +224,37 @@ class RealTimeClient {
    * @param {Object} filters Filters to be passed for the created subscription.
    * @param {function} handlerFunc The function to fire when updates are received for this
    * subscription.
-   * @returns {Promise} A promise that resolves when the subscription request has been sent.
+   * @returns {Promise} A promise that resolves when the subscription request has been sent. The
+   * value of the resolved promise is a function that will end the subscription.
    */
   startSubscription(entity, customerCode, filters, handlerFunc) {
-    // TODO: these subscriptions don't actually work yet.  Still need to correlate subscription
-    // success messages received from the server with our request IDs and handle appropriately.
-    // subscriptions will be implemented in an upcoming story.
     const message = messages.creators.createSubscriptionRequest(entity, customerCode, filters);
-    this.openRequests[message.request_id] = handlerFunc;
-    return this.sendMessage(message);
+    let subscriptionStartResolver;
+    let subscriptionStartRejecter;
+    const subscriptionStart = new Promise((resolve, reject) => {
+      subscriptionStartResolver = resolve;
+      subscriptionStartRejecter = reject;
+    });
+    this.openRequests[message.request_id] = {
+      handler: handlerFunc,
+      subscriptionStartResolver,
+      subscriptionStartRejecter,
+    };
+    return this.sendMessage(message)
+      .then(() => subscriptionStart)
+      .then(success => this.stopSubscription.bind(this, success.subscription_id));
+  }
+
+  /**
+   * Ends a subscription to the Track Real Time API.
+   * @param {string} subscriptionId Identity of the subscription
+   * @returns {void}
+   */
+  stopSubscription(subscriptionId) {
+    this.sendMessage({
+      type: messages.SUBSCRIPTION_END.REQUEST,
+      subscription_id: subscriptionId,
+    });
```

Comment:
Would it make sense to get rid of the subscription handler here? It would mean getting some messages from the server that won't be processed, but the client has already asked to stop receiving the messages anyway.

Reply:
Good point. I'll do that here.

Reply:
In reconsidering this, I think I'd prefer to keep the subscription handling termination in the message handling logic. That gives me the option to fail the subscription end request, even if it is very unlikely.